src/main/java/com/iexec/core/replicate/ReplicatesService.java

```diff
@@ -258,52 +360,29 @@ public void updateReplicateStatus(String chainTaskId,
     @Retryable(value = {OptimisticLockingFailureException.class}, maxAttempts = 100)
     public Optional<TaskNotificationType> updateReplicateStatus(String chainTaskId,
                                                                 String walletAddress,
-                                                                ReplicateStatusUpdate statusUpdate) {
+                                                                ReplicateStatusUpdate statusUpdate,
+                                                                UpdateReplicateStatusArgs updateReplicateStatusArgs) {
         log.info("Replicate update request [status:{}, chainTaskId:{}, walletAddress:{}, details:{}]",
                 statusUpdate.getStatus(), chainTaskId, walletAddress, statusUpdate.getDetailsWithoutStdout());
 
-        Optional<ReplicatesList> oReplicateList = getReplicatesList(chainTaskId);
-        if (oReplicateList.isEmpty() || oReplicateList.get().getReplicateOfWorker(walletAddress).isEmpty()) {
-            log.error("Cannot update replicate, could not get replicate [chainTaskId:{}, UpdateRequest:{}]",
-                    chainTaskId, statusUpdate);
+        if (!Objects.equals(
+                    canUpdateReplicateStatus(chainTaskId, walletAddress, statusUpdate, updateReplicateStatusArgs),
```

Comment:
Do we really need Objects.equals?

Reply:
You're right. As we compare enum member, a simple `==` would work.